tools/lkl/tests/net.sh

```diff
@@ -1,102 +1,175 @@
-    exit 0
-fi
+script_dir=$(cd $(dirname ${BASH_SOURCE:-$0}); pwd)
 
-# android doesn't have sudo
-if [ -z ${LKL_ANDROID_TEST} ] ; then
-    SUDO="sudo"
-fi
+source $script_dir/test.sh
+source $script_dir/net-setup.sh
+
+cleanup_backend()
+{
+    set -e
+
+    case "$1" in
+    "tap")
+        tap_cleanup
+        ;;
+    "pipe")
+        rm -rf $work_dir
+        ;;
+    "raw")
+        ;;
+    "macvtap")
+        sudo ip link del dev $(tap_ifname) type macvtap
+        ;;
+    "loopback")
+        ;;
+    esac
+}
 
-TEST_HOST=8.8.8.8
-IFNAME=`ip route get ${TEST_HOST} |head -n1 | cut -d ' ' -f5`
-GW=`ip route get ${TEST_HOST} |head -n1 | cut -d ' ' -f3`
+get_test_ip()
+{
+    # DHCP test parameters
+    TEST_HOST=8.8.8.8
+    HOST_IF=$(lkl_test_cmd ip route get $TEST_HOST | head -n1 |cut -d ' ' -f5)
+    HOST_GW=$(lkl_test_cmd ip route get $TEST_HOST | head -n1 | cut -d ' ' -f3)
+    if lkl_test_cmd ping -c1 -w1 $HOST_GW; then
+        TEST_IP_REMOTE=$HOST_GW
+    elif lkl_test_cmd ping -c1 -w1 $TEST_HOST; then
+        TEST_IP_REMOTE=$TEST_HOST
+    else
+        echo "could not find remote test ip"
+        return $TEST_SKIP
+    fi
 
-script_dir=$(cd $(dirname ${BASH_SOURCE:-$0}); pwd)
-cd ${script_dir}
-
-# Make a temporary directory to run tests in, since we'll be copying
-# things there.
-work_dir=$(mktemp -d)
-
-# And make sure we clean up when we're done
-function clear_work_dir {
-    rm -rf ${work_dir}
-    ${SUDO} ip link set dev lkl_ptt1 down &> /dev/null || true
-    ${SUDO} ip tuntap del dev lkl_ptt1 mode tap &> /dev/null || true
-    ${SUDO} ip link del dev lkl_vtap0 type macvtap &> /dev/null || true
+    export_vars HOST_IF TEST_IP_REMOTE
 }
 
-trap clear_work_dir EXIT
-
-echo "== PIPE (LKL net) tests =="
-if [ -z `which mkfifo` ]; then
-    echo "WARNIG: no mkfifo command, skipping PIPE tests."
-else
-
-fifo1=${work_dir}/fifo1
-fifo2=${work_dir}/fifo2
-mkfifo ${fifo1}
-mkfifo ${fifo2}
-hijack_script=${script_dir}/../bin/lkl-hijack.sh
-LKL_HIJACK_NET_IFTYPE=pipe \
-	LKL_HIJACK_NET_IFPARAMS="${fifo1}|${fifo2}" \
-	LKL_HIJACK_NET_IP=192.168.16.1 \
-	LKL_HIJACK_NET_NETMASK_LEN=24 \
-	${hijack_script} sleep 10 &
-sleep 5
-./net-test pipe "${fifo2}|${fifo1}" 192.168.16.1 192.168.16.2 24
-wait
-fi
+setup_backend()
+{
+    set -e
 
-echo "== TAP (LKL net) tests =="
-if [ -c /dev/net/tun ]; then
-    ${SUDO} ip link set dev lkl_ptt1 down || true
-    ${SUDO} ip tuntap del dev lkl_ptt1 mode tap || true
-    ${SUDO} ip tuntap add dev lkl_ptt1 mode tap user $USER
-    ${SUDO} ip link set dev lkl_ptt1 up
-    ${SUDO} ip addr add dev lkl_ptt1 192.168.14.1/24
+    if [ "$LKL_HOST_CONFIG_POSIX" != "y" ] &&
+       [ "$1" != "loopback" ]; then
+        echo "not a posix environment"
+        return $TEST_SKIP
+    fi
 
-    ./net-test tap lkl_ptt1 192.168.14.1 192.168.14.2 24
+    case "$1" in
+    "loopback")
+        ;;
+    "pipe")
+        if [ -z $(lkl_test_cmd which mkfifo) ]; then
```

Comment:
do you see why x86_64 test skips pipe test ?

https://circleci.com/gh/tavip/linux/151?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Reply:
There was an issue in the test, I've fixed it and will push a new version soon.

Reply:
thanks, will look at your new version.